query/src/main/scala/filodb/query/exec/rangefn/RangeFunction.scala

```diff
@@ -161,6 +161,44 @@ trait CounterChunkedRangeFunction[R <: MutableRowReader] extends ChunkedRangeFun
                     startTime: Long, endTime: Long): Unit
 }
 
+/**
+ * A ChunkedRangeFunction for period-counters/histograms. This does not handle resets/corrections.
+ * Data is assumed to be ordered.
+ */
+trait PeriodCounterChunkedRangeFunction[R <: MutableRowReader] extends ChunkedRangeFunction[R] {
+
+  // reset is called before first chunk.
+  override def reset(): Unit = {}
+
+  // scalastyle:off parameter.number
+  def addChunks(tsVectorAcc: MemoryReader, tsVector: BinaryVectorPtr, tsReader: bv.LongVectorDataReader,
```

Comment:
Is this pulled from `TimeRangeFunction`? Can it this extend that class, instead? (I see the code is a bit different, so it might not be a reasonable change-- I'm just looking to cut down on the # of places the `binarySearch`/`0x7fffffff`/`start`/`end` incantation shows up)